src/Microsoft.ComponentDetection.Detectors/pip/SimplePythonResolver.cs

```diff
@@ -0,0 +1,374 @@
+    /// <param name="state"> The PythonResolverState. </param>
+    /// <param name="spec"> The PipDependencySpecification. </param>
+    /// <returns> Returns a list of PipDependencySpecification. </returns>
+    private async Task<IList<PipDependencySpecification>> FetchPackageDependenciesAsync(
+    PythonResolverState state,
+    PipDependencySpecification spec)
+    {
+        var candidateVersion = state.NodeReferences[spec.Name].Value.Version;
+
+        var packageToFetch = state.ValidVersionMap[spec.Name][candidateVersion].FirstOrDefault(x => string.Equals("bdist_wheel", x.PackageType, StringComparison.OrdinalIgnoreCase)) ??
+                             state.ValidVersionMap[spec.Name][candidateVersion].FirstOrDefault(x => string.Equals("bdist_egg", x.PackageType, StringComparison.OrdinalIgnoreCase));
+        if (packageToFetch == null)
+        {
+            return new List<PipDependencySpecification>();
+        }
+
+        var packageFileStream = await this.simplePypiClient.FetchPackageFileStreamAsync(packageToFetch.Url);
+
+        if (packageFileStream.Length == 0)
+        {
+            return new List<PipDependencySpecification>();
+        }
+
+        return await this.FetchDependenciesFromPackageStreamAsync(spec.Name, candidateVersion, packageFileStream);
+    }
+
+    /// <summary>
+    /// Given a package stream will unzip and return the dependencies in the metadata file.
+    /// </summary>
+    /// <param name="name"> The package name. </param>
+    /// <param name="version"> The package version. </param>
+    /// <param name="packageStream"> The package file stream. </param>
+    /// <returns> Returns a list of the dependencies. </returns>
+    private async Task<IList<PipDependencySpecification>> FetchDependenciesFromPackageStreamAsync(string name, string version, Stream packageStream)
+    {
+        var dependencies = new List<PipDependencySpecification>();
+        var package = new ZipArchive(packageStream);
+
+        var entry = package.GetEntry($"{name.Replace('-', '_')}-{version}.dist-info/METADATA");
+
+        // If there is no metadata file, the package doesn't have any declared dependencies
+        if (entry == null)
+        {
+            return dependencies;
+        }
+
+        var content = new List<string>();
+        using (var stream = entry.Open())
+        {
+            using var streamReader = new StreamReader(stream);
+
+            while (!streamReader.EndOfStream)
+            {
+                var line = await streamReader.ReadLineAsync();
+
+                if (PipDependencySpecification.RequiresDistRegex.IsMatch(line))
+                {
+                    content.Add(line);
+                }
+            }
+        }
+
+        // Pull the packages that aren't conditional based on "extras"
+        // Right now we just want to resolve the graph as most comsumers will
+        // experience it
+        foreach (var deps in content.Where(x => !x.Contains("extra ==")))
```

Comment:
Is `extra==` also valid?

Reply:
Not sure, but the lines with the `extra==` seem to be valid packages but are conditional/optional which is why we skip them. I found an example in the metadata file image below. It shows botocore is already required and then botocore[crt] is an extra
![image](https://github.com/microsoft/component-detection/assets/8052624/4d59a70a-eeed-409b-a618-13ca85a9238b)

Reply:
I think this check needs to be more resilient. I've seen a variety of packages with strangely formatted `Requires-Dist`. I think in the Python Version class we have some regexes to parse parts of these out.

Reply:
Still a bit unclear. The Requires-Dist lines are still parsed with the `RequiresDistRegex` in `PipDependencySpecification` 
`Requires-Dist:\s*([^\s;\[<>=!~]+)(?:\[[^\]]+\])?(?:\s*\(([^)]+)\))?([^;]*)`

Reply:
Discussed offline, looks fine.